subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -364,7 +368,11 @@ static int try_resolve_literal_addr(const char *host, const char *service,
 	int protocol = IPPROTO_TCP;
 
 	if (!host) {
-		return DNS_EAI_NONAME;
+		if (hints->ai_flags & AI_PASSIVE) {
+			return getaddrinfo_null_host(atoi(service), hints, res);
```

Comment:
Please fix: instead of converting the port to `int` here you should bring the port conversion and validity check that's [below](https://github.com/DynonAvionics/zephyr/pull/5/files#diff-e626ddeee4564ffa317b0ef5c675200d34fbfbf3494fa7d8eb294656543a1bb6R398) above this `if`.